frame/support/src/storage/mod.rs

```diff
@@ -25,6 +25,22 @@ pub mod hashed;
 pub mod child;
 pub mod generator;
 
+/// Execute under a transactional layer.
+///
+/// If the result of execution is an error,
+/// the transactional layer get reverted; otherwhise
+/// it is committed.
+pub fn with_transaction<R, E>(f: impl FnOnce() -> Result<R, E>) -> Result<R, E> {
+	sp_io::storage::start_transaction();
+	let r = f();
```

Comment:
@cheme what do you expect to gain by assigning from one single-letter-function call to a single-letter-variable?

```suggestion
	let result = f();
```